src/socket/address.cr

```diff
@@ -227,7 +227,9 @@ class Socket
       in LibC::InAddr
         addr.s_addr & 0x000000ff_u32 == 0x0000007f_u32
       in LibC::In6Addr
-        ipv6_addr8(addr) == StaticArray[0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 1_u8]
+        addr8 = ipv6_addr8(addr)
+        addr8 == StaticArray[0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 1_u8] || # "::1"
+          addr8.to_slice[0, 13] == Bytes[0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 255_u8, 255_u8, 127_u8] # "::ffff:127.0.0.1/104"
```

Comment:
This should be turned into a constant to avoid having to allocate the `Bytes` on every call

Reply:
Or StaticArray#to_slice? will only stack allocate.

Reply:
I think it might be simpler to turn the raw bytes into `UInt64` or whatever and compare things against numbers. It should also be faster.

Reply:
Same speed for `StaticArray[...]#to_slice` and `.unsafe_as(UInt128)` (1ns to do the check), I guess it comes down to readability. See for comparison: https://github.com/crystal-lang/crystal/pull/12783/commits/3663384967235c99e6f02016fc92fcf0ae24f045